fe/fe-core/src/main/java/org/apache/doris/nereids/parser/NereidsParser.java

```diff
@@ -290,9 +292,37 @@ public Optional<String> parseForSyncMv(String sql) {
         return logicalPlanBuilderForSyncMv.getQuerySql();
     }
 
+    /** get hint map */
+    public static Map<Integer, ParserRuleContext> getHintMap(String sql,
+                                                             Function<DorisParser, ParserRuleContext> parseFunction) {
+        // parse hint first round
+        DorisLexer hintLexer = new DorisLexer(new CaseInsensitiveStream(CharStreams.fromString(sql)));
+        hintLexer.setChannel2(true);
+        CommonTokenStream hintTokenStream = new CommonTokenStream(hintLexer);
```

Comment:
antlr could generate one stream and get specific channel token from it by `getHiddenTokensToRight`, so could we use it to get hint in `visitRegularQuerySpecification`?

Reply:
getHiddenTokensToRight does not work in this case, I tried to get it but it failed. I try to use List<Token> hintTokens = hintTokenStream.getHiddenTokensToRight(0, 2); but it failed to get token list out from channel2